### PR TITLE
[MM-20378] Convert file_attachment_list and tests to TypeScript

### PR DESCRIPTION
**Pull request by LauSam09**
#### Summary
This pull request converts the `file_attachment_list` component and tests to TypeScript.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/16729
JIRA: https://mattermost.atlassian.net/browse/MM-20378

### CODE REVIEW

File: components/file_attachment_list/file_attachment_list.test.tsx [hunk 1]
```diff
@@ -4,17 +4,57 @@
...
+import {Post} from 'mattermost-redux/src/types/posts';
+
 import FileAttachment from 'components/file_attachment';
 import SingleImageView from 'components/single_image_view';
 
 import FileAttachmentList from './file_attachment_list';
 
 describe('FileAttachmentList', () => {
-    const post = {id: 'post_id', file_ids: ['file_id_1', 'file_id_2', 'file_id_3']};
+    const post: Post = {
```
Comment (devinbinnie) [inline components/file_attachment_list/file_attachment_list.test.tsx:15]: For future reference, we have a `getPostMock` function in `utils/test_helper` that mocks out all the extra data you need here.

File: components/file_attachment_list/index.ts [hunk 2]
```diff
@@ -6,15 +6,16 @@ import {connect} from 'react-redux';
...
 import {isEmbedVisible} from 'selectors/posts';
 
-import FileAttachmentList from './file_attachment_list.jsx';
+import FileAttachmentList, {Props} from './file_attachment_list';
 
 function makeMapStateToProps() {
     const selectFilesForPost = makeGetFilesForPost();
 
-    return function mapStateToProps(state, ownProps) {
+    return function mapStateToProps(state: GlobalState, ownProps: Props) {
```
Comment (devinbinnie) [inline components/file_attachment_list/index.ts:18]: It's generally not a good practice to use the same `Props` type that the component uses for the redux connector.
What will happen here is that any component that consumes this one will expect all props that are required for the component (eg. `locale`) to be explicitly provided by the consuming component, causing some annoying type errors.

I'd recommend here creating an `OwnProps` type specifically for what the `mapStateToProps` function needs to fetch other data and using that type instead.

**Review by devinbinnie — CHANGES_REQUESTED** [review]
Thanks for the contribution @LauSam09!
Looks good overall, just had a couple small comments to resolve.

File: components/file_attachment_list/index.ts (hunk 2 above)
Comment (LauSam09) [reply components/file_attachment_list/index.ts]: Thanks for the explanation - I've made the change you suggested.

File: components/file_attachment_list/file_attachment_list.test.tsx (hunk 1 above)
Comment (LauSam09) [reply components/file_attachment_list/file_attachment_list.test.tsx]: Handy - thanks!

File: components/file_attachment_list/file_attachment_list.test.tsx [hunk 3]
```diff
@@ -32,7 +37,21 @@ describe('FileAttachmentList', () => {
             ...baseProps,
             fileCount: 1,
             fileInfos: [
-                {id: 'file_id_1', name: 'file.txt', extension: 'txt'},
+                {
```
Comment (devinbinnie) [inline components/file_attachment_list/file_attachment_list.test.tsx:40]: Could use `getFileInfoMock` here too if you wanted

**Review by devinbinnie — APPROVED** [review]
Thanks @LauSam09, this looks great now! Just one non-blocking comment.

File: components/file_attachment_list/file_attachment_list.test.tsx (hunk 3 above)
Comment (LauSam09) [reply components/file_attachment_list/file_attachment_list.test.tsx]: Oops - done!

**Review by hmhealey — APPROVED** [review]
Thanks, @LauSam09. Looks good to me!

**Review by jgilliam17 — APPROVED** [review]
Thank you @LauSam09 
Tested, looks good to merge.
- Verified file attachments and previews - as before.
@hmhealey Can you please help merge? Thanks 🙂

**Pull request merged**